src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -236,6 +236,9 @@ public bool IsMarshallingRequired()
             return true;
         }
 
+        /// <summary>
+        /// Given a NativeType enum returns a corresponding Type that should be passed to the Native side
```

Comment:
This does not match what the implementation does. E.g. It returns managed `char[]` for LPStr. You are not passing `char[]` to native side for `LPStr` - you are passing `char*`.